### PR TITLE
Update Graph Class, move /graph_search from planners to utils.

### PR DESCRIPTION
**Pull request by Srujan-D**
1) Add method in Graph class (gennav.utils.graph) to delete edge connecting given two nodes from the graph.
2) Move /graph_search from gennav.planners to gennav.utils.

### CODE REVIEW

File: gennav/utils/graph.py [hunk 1]
```diff
@@ -33,6 +33,32 @@ def add_edge(
...
 
+    def del_edge(self, from_node, to_node):
+        """Deletes edge connecting two nodes to the graph.
+
+        Args:
+            from_node (gennav.utils.common.Node): starting node of the edge.
+            to_node (gennav.utils.common.Node): ending node of the edge.
+        """
+
+        if len(self.edges[from_node]):
```
Comment (threewisemonkeys-as) [inline gennav/utils/graph.py]: Can you make this if .. == 0:

Makes it more explicit

File: gennav/utils/graph.py [hunk 2]
```diff
@@ -33,6 +33,32 @@ def add_edge(
...
+            if (len(self.edges[from_node])) == 1:
+                del self.edges[from_node]
+            else:
+                self.edges[from_node].remove(to_node)
+
+            del self.distances[(from_node, to_node)]
+        else:
+            print("Edge does not exist")
```
Comment (threewisemonkeys-as) [inline gennav/utils/graph.py]: You could raise ValueError here

**Review by threewisemonkeys-as — CHANGES_REQUESTED** [review]

File: gennav/utils/graph.py (hunk 1 above)
Comment (Srujan-D) [reply gennav/utils/graph.py]: Done.

File: gennav/utils/graph.py (hunk 2 above)
Comment (Srujan-D) [reply gennav/utils/graph.py]: Done.

**Comment by threewisemonkeys-as** [comment]
Could you look at the pytest errors?

**Comment by Srujan-D** [comment]
> Could you look at the pytest errors?

Done, they were about updating the astar code to suit the modified Node class.

File: gennav/utils/graph.py [hunk 3]
```diff
@@ -33,6 +33,32 @@ def add_edge(
...
+        else:
+            if len(self.edges[from_node]) == 1:
+                del self.edges[from_node]
+            else:
+                self.edges[from_node].remove(to_node)
+
+            del self.distances[(from_node, to_node)]
+
+        if len(self.edges[to_node]) == 0:
+            raise ValueError
```
Comment (threewisemonkeys-as) [inline gennav/utils/graph.py]: Add string describing what error is

**Review by threewisemonkeys-as — CHANGES_REQUESTED** [review]

File: gennav/utils/graph.py (hunk 3 above)
Comment (Srujan-D) [reply gennav/utils/graph.py]: Done.

**Comment by threewisemonkeys-as** [comment]
Should I merge this or are you going to make the changes to PRM on this PR itself?

**Comment by Srujan-D** [comment]
> Should I merge this or are you going to make the changes to PRM on this PR itself?

If you want different PR for those changes, you can merge this PR.
I have done the necessary changes to the concerned codes, but the visualize graph isn't working at all. The builds are passing whenever I don't include the visualization lines in the test file. I need some time to figure out what's causing the fail.

**Comment by threewisemonkeys-as** [comment]
We can leave the visualisation for later. Lets get the core functionality done. I'll merge this. You can open another PR for the other changes.

**Pull request merged**